lib/index.js

```diff
@@ -1,343 +1,381 @@
+                                return internals.skippedCols.indexOf(skippedInfo) === i;
+                            });
+                            skippedColsMsg = `Skipped unsupported columns:${Os.EOL}${dedupedSkippedCols.join(Os.EOL)}`;
+                        }
 
-                const compiler = Handlebars.compile(String(Fs.readFileSync(__dirname + '/hbs/migrationFileTemplate.hbs')));
-                const createdTables = tableDeltas.tableDelta.create.concat(tableDeltas.joinTableDelta.create);
+                        if (createdTables.length === 0 &&
+                            columnDeltas.changes.length === 0) {
 
-                if (createdTables.length === 0 &&
-                    columnDeltas.changes.length === 0) {
+                            const noMigrationMsg = !!skippedColsMsg ? `${skippedColsMsg}${Os.EOL}No migration needed` : 'No migration needed';
+                            return cb(null, noMigrationMsg);
+                        }
 
-                    return cb(null, 'No migration needed');
-                }
+                        const migrationFileContents = compiler({
+                            allTables,
+                            createdTables,
+                            tableDeltas,
+                            columnDeltas
+                        });
 
-                const migrationFileContents = compiler({
-                    allTables,
-                    createdTables,
-                    tableDeltas,
-                    columnDeltas
-                });
+                        Promise.resolve(migrationGroup.knex.migrate.make(userOptions.migrationName || 'schwifty-migration', {
+                            directory: migrationGroup.migrationsDir
+                        }))
+                            .asCallback((err) => {
 
-                migrationGroup.knex.migrate.make(userOptions.migrationName || 'schwifty-migration', {
-                    directory: migrationGroup.migrationsDir
-                })
-                    .then(() => {
+                                if (err) {
+                                    return cb(err);
+                                }
 
-                        const migrationsDirFiles = Fs.readdirSync(migrationsDir);
-                        const justCreatedMigration = migrationsDirFiles[migrationsDirFiles.length - 1];
-                        const justCreatedMigrationPath = Path.resolve(migrationsDir, justCreatedMigration);
-                        Fs.writeFileSync(justCreatedMigrationPath, migrationFileContents);
+                                const migrationsDirFiles = Fs.readdirSync(migrationsDir);
+                                const justCreatedMigration = migrationsDirFiles[migrationsDirFiles.length - 1];
+                                const justCreatedMigrationPath = Path.resolve(migrationsDir, justCreatedMigration);
+                                Fs.writeFileSync(justCreatedMigrationPath, migrationFileContents);
```

Comment:
Maybe we could make this not-sync.

Reply:
I did this -- now trying to concoct a way to hit that error in tests -- the migrationsDir is already touched by knex when the file is created, so we can't change write perms on a dir -- using a too-long filename won't work because same.. it's a brain buster

Reply:
Would suggest monkey-patching `Fs.writeFile` then restoring it when you're done.